### PR TITLE
fix: get BOM while creating wo on the basis of bom item

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
Ref: [Asana](https://app.asana.com/0/1192407896547052/1199584324763283)

### CODE REVIEW

File: erpnext/stock/doctype/purchase_receipt/purchase_receipt.py [hunk 1]
```diff
@@ -838,3 +832,20 @@ def make_work_orders(items, purchase_receipt, company, project=None):
...
+				bom.name
+			FROM
+				`tabBOM` as bom
+			INNER JOIN
+				`tabBOM Item` as bom_item
+			ON
+				bom_item.item_code = (%s)
+			WHERE
+				bom.name = bom_item.parent
+		""",filters.get("item_code"))
```
Comment (hrwX) [inline erpnext/stock/doctype/purchase_receipt/purchase_receipt.py]: Use ORM

File: erpnext/stock/doctype/purchase_receipt/purchase_receipt.py [hunk 2]
```diff
@@ -838,3 +832,20 @@ def make_work_orders(items, purchase_receipt, company, project=None):
 		out.append(work_order)
 
 	return [p.name for p in out]
+
+@frappe.whitelist()
+def get_bom_query(doctype, txt, searchfield, start, page_len, filters):
```
Comment (hrwX) [inline erpnext/stock/doctype/purchase_receipt/purchase_receipt.py:835]:
```suggestion
def get_bom_query(doctype, filters):
```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/purchase_receipt/purchase_receipt.py (hunk 2 above)
Comment (MyuddinKhatri) [reply erpnext/stock/doctype/purchase_receipt/purchase_receipt.py:835]: As the function is called in get_query method definition requires this, it gives error without this args

File: erpnext/stock/doctype/purchase_receipt/purchase_receipt.py [hunk 3]
```diff
@@ -838,3 +832,20 @@ def make_work_orders(items, purchase_receipt, company, project=None):
...
+			FROM
+				`tabBOM` as bom
+			INNER JOIN
+				`tabBOM Item` as bom_item
+			ON
+				bom_item.item_code = (%s)
+			WHERE
+				bom.manufacturing_type = (%s)
+				AND bom.name = bom_item.parent
+		""",(filters.get("item_code"), filters.get("manufacturing_type")))
```
Comment (hrwX) [inline erpnext/stock/doctype/purchase_receipt/purchase_receipt.py]:
```suggestion
	boms = frappe.get_all("BOM", filters=[
		["BOM Item", "item_code", "=", filters.get("item_code")],
		["BOM", "manufacturing_type", "=", filters.get("manufacturing_type")]
	], fields=["`tabBOM`.name"])
```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/purchase_receipt/purchase_receipt.py [hunk 4]
```diff
@@ -515,10 +514,6 @@ def get_work_order_items(self, for_raw_material_request=0):
 
 		for table in [self.items, self.supplied_items]:
 			for row in table:
-				bom = get_default_bom_item(row.item_code)
-				if not bom:
-					continue
-				
 				stock_qty = row.consumed_qty if row.doctype == 'Purchase Receipt Item Supplied' else row.stock_qty
 				pending_qty = stock_qty
 				if not for_raw_material_request:
```
Comment (vishdha) [inline erpnext/stock/doctype/purchase_receipt/purchase_receipt.py]: @MyuddinKhatri  this condition will always be false. you are not sending any args from the calling function

**Review by vishdha — CHANGES_REQUESTED** [review]

**Review by vishdha — CHANGES_REQUESTED** [review]
@MyuddinKhatri  When select box make sure its finished good available in the dialog box. Add BOM Finished good to production item in the work order. currently irrespective of BOM it set finished good and raw material with the same item.
at line 818 in purchase recipt make sure it goes bom finished good item

**Review by hrwX — APPROVED** [review]

**Review by vishdha — APPROVED** [review]

**Pull request merged**